docs/practices/pull_requests.md

```diff
@@ -0,0 +1,84 @@
+# Pull request workflow
+
+Our team practices should encourage transparency and collaboration, and reviewing one another's work is a great way to ensure that we share context and knowledge, and improve the codebase.
+Our goals are to balance inclusive and transparent team practices with the reality that we are all asynchronous and don't want to pay a large coordination penalty for every change and decision.
+
+**All changes to a JupyterHub repository should happen via Pull Requests**.
+Any exceptions to this should be explicitly listed in team documentation.[^1]
+
+[^1]: The only exception we can think of is when pushing a commit to `main` during a release, but we'll leave this open-ended just in case there are other cases where this is useful.
+
+(prs:principles)=
+## Principles to follow
+
+- **Provide context**.
+  Provide enough discussion and context sharing so that every team member understands the rationale for a change, even if they may not personally agree with it.
+  A Pull Request should link to (and optionally close) any locations where discussion happened.
+  This gives each change more provenance and separates design from implementation.
+  The more complex or impactful a change is, the more time should be given to discuss and include the perspective of others.
+  Discussions can happen in issues, pull requests, or in another space that is accessible to non-coders (like a forum post).
+  Where to have these discussions depends on the type, scope, and size of the change that's being proposed, use your best judgement (and see the next point).
+- **Include others**.
+  Actively invite participation and comments from others, especially if you think they are interested in the proposed change.
+  Go out of your way to make it easy for others to participate, and make your best effort at incorporating suggestions and feedback from others.
+  For discussions that are complex, potentially controversial, or have an uncertain resolution, open an issue or forum post to discuss with others before diving into a code implementation.
+  Consider the skills and comfort level of others in the discussion, and choose a location and format for the discussion that will be accessible to all relevant stakeholders.
```

Comment:
Not sure if this is equivalent :-/ It is an attempt at saying "We want you to want others to take part in discussions, think about where the people with the skills are, go there. Don't expect them to come to you."
```suggestion
  Consider the skills and comfort level of the people who you want to take part in the discussion. Choose a location and format for the discussion that will be accessible to them. Think about where those with expertise on the topic can be reached, not where is easiest for you.
```